framework/util/image_writer.cpp

```diff
@@ -25,26 +25,57 @@
 
 #include "platform.h"
 
+#if defined(ENABLE_ZLIB_COMPRESSION) && defined(ENABLE_PNG_SCREENSHOT)
+#include <zlib.h>
+
+// This function re-formats the call to the Zlib compress2 function so that we can
+// use it in the STB PNG generation code.  Using the default STB PNG compression
+// resulted in a 20% reduction versus the original image in tests, while using the
+// zlib compress2 function resulted in a 40+% reduction versus the original image.
+uint8_t* gfxr_zlib_compress2(uint8_t* data, int32_t data_len, int32_t* out_len, int32_t quality)
+{
+    uint8_t*      target  = reinterpret_cast<uint8_t*>(malloc(data_len));
```

Comment:
I know typical platforms fail allocation lazily but it would be easy to check malloc's return value here since we already have a return path returning a null that the caller is expected to deal with.
`compress2` doesn't seem to have an immediate null check of its own (https://github.com/luvit/zlib/blob/master/compress.c#L38).

Reply:
Done.